src/UnitfulRecipes.jl

```diff
@@ -27,42 +29,41 @@ function fixaxis!(attr, x, axisletter)
     axislims = Symbol(axisletter, :lims)   # xlims, ylims, zlims
     axisticks = Symbol(axisletter, :ticks) # xticks, yticks, zticks
     err = Symbol(axisletter, :error)       # xerror, yerror, zerror
-    axisunit = Symbol(axisletter, :unit)   # xunit, yunit, zunit
-    axis = Symbol(axisletter, :axis)       # xaxis, yaxis, zaxis
-    # Get the unit
-    u = pop!(attr, axisunit, unit(eltype(x)))
-    # If the subplot already exists with data, get its unit
-    sp = get(attr, :subplot, 1)
-    if sp ≤ length(attr[:plot_object]) && attr[:plot_object].n > 0
-        label = attr[:plot_object][sp][axis][:guide]
-        if label isa UnitfulString
-            u = label.unit
-        end
-        # If label was not given as an argument, reuse
-        get!(attr, axislabel, label)
-    end
+
+    u = fixlabel!(attr, axisletter, unit(first(x)))
     # Fix the attributes: labels, lims, ticks, marker/line stuff, etc.
     append_unit_if_needed!(attr, axislabel, u)
     ustripattribute!(attr, axislims, u)
     ustripattribute!(attr, axisticks, u)
     ustripattribute!(attr, err, u)
-    fixmarkercolor!(attr)
+    fixcolors!(attr)
     fixmarkersize!(attr)
-    fixlinecolor!(attr)
     # Strip the unit
-    ustrip.(u, x)
+    return ustrip.(u, x)
 end
 
-# Recipe for (x::AVec, y::AVec, z::Surface) types
-const AVec = AbstractVector
-const AMat{T} = AbstractArray{T,2} where T
-@recipe function f(x::AVec, y::AVec, z::AMat{T}) where T <: Quantity
-    u = get(plotattributes, :zunit, unit(eltype(z)))
-    ustripattribute!(plotattributes, :clims, u)
-    z = fixaxis!(plotattributes, z, :z)
-    append_unit_if_needed!(plotattributes, :colorbar_title, u)
-    x, y, z
```

Comment:
I don't think we need this recipe if `x`, `y` and `z` are correctly identified as type targets for themselves.